spec/controllers/service_controller_spec.rb

```diff
@@ -378,5 +378,30 @@
     end
   end
 
+  context 'clicking on Active/Retired Services node in the tree, after applying a filter' do
+    describe '#get_node_info' do
+      before do
+        allow(controller).to receive(:session).and_return(:adv_search => {"Service" => {:expression => {}}}, :edit => {:expression => {}})
```

Comment:
Could you make this a little bit more readable? Maybe expand it into multiple lines? Or just my screen is too small? :wink: